wp-includes/class-wp-https-ui.php

```diff
@@ -0,0 +1,336 @@
+
+	/**
+	 * The ID of the HTTPS settings section.
+	 *
+	 * @var string
+	 */
+	const HTTPS_SETTING_ID = 'wp_upgrade_https';
+
+	/**
+	 * The maximum number of URLs that show initially, before clicking 'Show more'.
+	 *
+	 * @var int
+	 */
+	const NUMBER_INITIAL_URLS = 4;
+
+	/**
+	 * The number of URLs in each <ul>, which display on clicking 'Show more'.
+	 *
+	 * @var string
+	 */
+	const NUMBER_URLS_IN_EACH_UL = 10;
+
+	/**
+	 * The max character length of the URL, after which it will be truncated with an ellipsis.
+	 *
+	 * @var int
+	 */
+	const MAX_URL_LENGTH = 75;
+
+	/**
+	 * The instance of WP_HTTPS_Detection.
+	 *
+	 * @var WP_HTTPS_Detection
+	 */
+	public $wp_https_detection;
+
+	/**
+	 * WP_HTTPS_UI constructor.
+	 *
+	 * @param WP_HTTPS_Detection $wp_https_detection An instance of WP_HTTPS_Detection.
+	 */
+	public function __construct( $wp_https_detection ) {
+		$this->wp_https_detection = $wp_https_detection;
+	}
+
+	/**
+	 * Initializes the object.
+	 */
+	public function init() {
+		add_action( 'admin_init', array( $this, 'init_admin' ) );
+		add_action( 'init', array( $this, 'filter_site_url_and_home' ) );
+		add_action( 'init', array( $this, 'filter_header' ) );
+	}
+
+	/**
+	 * Initializes the admin tasks.
+	 */
+	public function init_admin() {
+		$this->register_settings();
+		$this->add_settings_field();
+	}
+
+	/**
+	 * Registers the HTTPS settings.
+	 */
+	public function register_settings() {
+		register_setting(
+			self::OPTION_GROUP,
+			self::UPGRADE_HTTPS_OPTION,
+			array(
+				'type'              => 'string',
+				'sanitize_callback' => array( $this, 'upgrade_https_sanitize_callback' ),
+			)
+		);
+	}
+
+	/**
+	 * Sanitization callback for the upgrade HTTPS option.
+	 *
+	 * @param string $raw_value The value to sanitize.
+	 * @return bool Whether the option is true or false.
+	 */
+	public function upgrade_https_sanitize_callback( $raw_value ) {
+		unset( $raw_value );
+		return isset( $_POST[ self::UPGRADE_HTTPS_OPTION ] ); // WPCS: CSRF OK.
+	}
+
+	/**
+	 * Adds a settings field to the 'General Settings' page.
+	 *
+	 * Only add this if the site can support HTTPS,
+	 * but the home and siteurl option values are not HTTPS (WordPress Address and Site Address).
+	 * This UI would not apply if those URLs are already HTTPS.
+	 */
+	public function add_settings_field() {
+		if ( ! $this->wp_https_detection->is_currently_https() ) {
+			add_settings_field(
+				self::HTTPS_SETTING_ID,
+				__( 'HTTPS', 'pwa' ),
+				array( $this, 'render_https_settings' ),
+				self::OPTION_GROUP
+			);
+		}
+	}
+
+	/**
+	 * Renders the HTTPS settings in /wp-admin on the General Settings page.
+	 */
+	public function render_https_settings() {
+		$upgrade_https_value = (bool) get_option( self::UPGRADE_HTTPS_OPTION );
+		$https_more_details  = sprintf(
+			'<a href="%s">%s</a>',
+			__( 'https://make.wordpress.org/support/user-manual/web-publishing/https-for-wordpress/', 'pwa' ),
+			__( 'More details', 'pwa' )
```

Comment:
Best to use `esc_html__()` here.

Reply:
Sure, https://github.com/xwp/pwa-wp/pull/36/commits/2a87852716f186ea3e83d182cd3856837fb2c035 uses `esc_html__()`.